File: frontend/src/types/fire.ts
export interface EncompassingRelationship {
  parent_card_id: string;
  child_card_id: string;
  weight: number;
}

export interface FIReData {
  relationships: EncompassingRelationship[];
  standalone_cards: string[];
}

export interface CreditSimulationResult {
  card_id: string;
  passed: boolean;
  credits: { card_id: string; credit: number }[];
}

File: backend/app/models/fire.py
from pydantic import BaseModel


class EncompassingRelationship(BaseModel):
    parent_card_id: str
    child_card_id: str
    weight: float


class FIReData(BaseModel):
    relationships: list[EncompassingRelationship]
    standalone_cards: list[str]


class CreditSimRequest(BaseModel):
    card_id: str
    passed: bool


class CreditSimResult(BaseModel):
    card_id: str
    passed: bool
    credits: list[dict]  # [{card_id, credit}]
